lib/security/security-spec.js

```diff
@@ -0,0 +1,70 @@
+var assert = require('assert');
+var sampleUserProfileData = require('../../test/fixtures/sampleUserProfile.json');
+var aes = require('./aes.js');
+var sha256 = require('./sha256.js');
+var sampleSecurityeData = require('../../test/fixtures/sampleSecurityData.json');
+
+describe("Test Data Encyption, Hashing and Decryption", function() {
+  it('should error when you try to hash a null value', function(done) {
+    return sha256.hash(null).then(function(response) {
+      assert(!response, "Should not contain a hash value,");
+    }).then(done).catch(function(err) {
+      assert(err, "should contain an error.");
+      done();
+    });
+  });
+
+  it('should respond with a correct hash when you try to hash a string value', function(done) {
+    sha256.hash(sampleSecurityeData.plaintext).then(function(response) {
+      assert.equal(response, sampleSecurityeData.plaintextSha256);
+    }).then(done);
+  });
+
+  it('should error when you try to encrypt using a null data value', function(done) {
+    aes.encrypt(null, sampleSecurityeData.aesKey).then(function() {
+      done(new Error('It should not try to encrypt using a null data value'));
+    }, function(err) {
```

Comment:
Using the second argument to `then(..)` to catch errors is generally regarded as an anti-pattern. The rules for catching errors in it are a little weird -- if the first argument (onSuccess function) throws an error for whatever reason, it will not trigger the second (onFailure) function, which is rarely what you want. Most people just recommend adding a `catch(...)` function after the then.

Now in the case of this function, that is kind of what you want, but not what I would recommend. (Also, the onFailure function would never be called if there wasn't an error, so asserting that there was an error is not necessary.) You really should be testing that you're getting the right error message on tests like this. That means adding a catch at the end and making sure that err.message is what you expect. (You don't want it to be failing by pure luck, like if you just had a typo somewhere.)

```js
return aes.encrypt(null, sampleSecurityData.aesKey)
  .thenReject(new Error('encrypt should have failed'))
  .catch(function(err) {
    assert.equal(err.message, 'Missing parameter foo');
  });
```

(The Bluebird promise library actually provides a nifty utility for this, where you can add a catch block but tell it to only catch errors with a specified message. It will let all others propagate through.)